src/modules/blocks/create-code-block.js

```diff
@@ -0,0 +1,81 @@
+import { toBlock } from '../utils/dom-utils.js';
+
+// Gets language class eg- language-html
+function getLanguageClass(element) {
+  let languageName;
+  if (element.classList) {
+    const classNames = Array.from(element.classList);
+    classNames.forEach((name) => {
+      if (name.startsWith('language-')) {
+        languageName = name;
+      }
+    });
+  }
+  return languageName || false;
+}
+
+export default function createCodeBlock(document) {
+  const codeElements = Array.from(document.querySelectorAll('pre > code'));
+  codeElements.forEach((element) => {
+    const blockOptions = {};
```

Comment:
can we make this into an array of strings (classes), then add to it the classes you want based on the data attributes.

Could also prefix each class with an `o-` and let the frontend parse it.

And better yet if you take this whole process in a separate function outside `createCodeBlock` and keep it Pure